README.markdown

```diff
@@ -449,6 +449,22 @@ If you want to stop processing jobs, but want to leave the worker running
 (for example, to temporarily alleviate load), use `USR2` to stop processing,
 then `CONT` to start it again.
 
+#### Signals on Heroku
+
+When shutting down processes, Heroku sends every process a TERM signal at the
+same time. By default this causes an immediate shutdown of any running job
+leading to frequent `Resque::TermException` errors.  For short running jobs, a simple
+solution is to give a small amount of time for the job to finish
+before killing it.
+
+To accomplish this set the following environment variables:
+
+* `RESQUE_PRE_SHUTDOWN_TIMEOUT` - The time between the parent receiving a shutdown signal (TERM by default) and it sending that signal on to the child process. Designed to give the child process
+time to complete before being forced to die.
+
+* `RESQUE_SHUTDOWN_SIGNAL` - The signal sent to the child process
```

Comment:
I honestly don't remember that well.  I wrote this code 2 years ago at this point.  I think it was to make the behavior backwards compatible.  Here's the blog post I wrote explaining the issue: http://eng.joingrouper.com/blog/2014/06/27/too-many-signals-resque-on-heroku/.  Based on re-reading that I think the child process needs to be able to ignore SIGTERM because heroku sends it that signal earlier than it should.